src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -232,9 +233,19 @@ private void loadEnvVariables(){
             this.collectBuildLogs = Boolean.valueOf(collectBuildLogsEnvVar);
         }
 
-        String collectBuildTracesEnvVar = System.getenv(COLLECT_BUILD_TRACES_PROPERTY);
-        if(StringUtils.isNotBlank(collectBuildTracesEnvVar)){
-            this.collectBuildTraces = Boolean.valueOf(collectBuildTraces);
+        String enableCiVisibilityVar = System.getenv(ENABLE_CI_VISIBILITY_PROPERTY);
+        if(StringUtils.isNotBlank(enableCiVisibilityVar)){
+            final boolean enableCiVisibility = Boolean.valueOf(enableCiVisibilityVar);
+            if(!enableCiVisibility || DatadogClient.ClientType.DSD.name().equals(this.reportWith)) {
+                this.collectBuildTraces = enableCiVisibility;
+            } else {
+                logger.warning("CI Visibility can only be enabled using Datadog Agent mode.");
```

Comment:
It seems this would warn if users set the env var `DATADOG_JENKINS_PLUGIN_ENABLE_CI_VISIBILITY` to `False`? Is that a bug?

Reply:
Hummm, no... The warning is shown when:
- `enableCiVisibility == true && DatadogClient.ClientType.HTTP.name().equals(this.reportWith)`

I'm going to change the condition to be clearer.